From 6d4677519856f5c33ad9ff6c3471ef126b8804cf Mon Sep 17 00:00:00 2001
From: Dhruva Gole <d-gole@ti.com>
Date: Tue, 20 Jun 2023 15:10:49 +0530
Subject: [PATCH 026/143] remoteproc: k3-m4: set as wakeup capable but keep
 disabled by default

commit 3c47db3916a0 ("remoteproc: k3-m4: Enable MCU wakeup support")
enabled k3-m4 rproc as a wakeup source, however let's keep it as wakeup
capable, but not wakeup enabled by default. The user is expected to
explicitly go and set this as enabled if they wish to use it as a wakeup
source.

Suggested-by: Vibhore Vardhan <vibhore@ti.com>
Signed-off-by: Dhruva Gole <d-gole@ti.com>
---
 drivers/remoteproc/ti_k3_m4_remoteproc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/remoteproc/ti_k3_m4_remoteproc.c b/drivers/remoteproc/ti_k3_m4_remoteproc.c
index 08e47e10b815..3c6f2fca5af4 100644
--- a/drivers/remoteproc/ti_k3_m4_remoteproc.c
+++ b/drivers/remoteproc/ti_k3_m4_remoteproc.c
@@ -431,7 +431,7 @@ static int k3_m4_rproc_probe(struct platform_device *pdev)
 
 	if (device_property_present(dev, "wakeup-source")) {
 		dev_dbg(dev, "registering as wakeup source\n");
-		device_init_wakeup(dev, true);
+		device_set_wakeup_capable(dev, true);
 	}
 
 	kproc->reset = devm_reset_control_get_exclusive(dev, NULL);
-- 
2.41.0

